### PR TITLE
Added query parameters options

### PR DESCRIPTION
**Pull request by matteo-campana**
# AzureCognitiveSearchRetriever integration with parameter options

This allows us to specify the query parameters to incorporate into the search query, making it possible to perform all the operations documented in the Microsoft site https://learn.microsoft.com/en-us/rest/api/searchservice/search-documents. 

Now, to avoid retrieving the entire contents of our index, we can write snippets like this to return only the top 3 documents

```python

query_parameters = {
    "top" : 3
}

retriever = AzureCognitiveSearchRetriever(content_key="content", query_parameters=query_parameters)

```

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

### CODE REVIEW

File: langchain/retrievers/azure_cognitive_search.py [hunk 1]
```diff
@@ -20,15 +20,42 @@ class AzureCognitiveSearchRetriever(BaseRetriever, BaseModel):
     index_name: str = ""
     """Name of Index inside Azure Cognitive Search service"""
     api_key: str = ""
-    """API Key. Both Admin and Query keys work, but for reading data it's
-    recommended to use a Query key."""
-    api_version: str = "2020-06-30"
+    """Required. Version of the REST API used for the request. For a list of supported versions, see API versions. For this operation, the api-version is specified as a URI parameter regardless of whether you call Search Documents with GET or POST."""
+    api_version: str = "2021-04-30-Preview"
```
Comment (dev2049) [inline langchain/retrievers/azure_cognitive_search.py:24]: mm would prefer to stick to stable version

File: langchain/retrievers/azure_cognitive_search.py [hunk 2]
```diff
@@ -20,15 +20,42 @@ class AzureCognitiveSearchRetriever(BaseRetriever, BaseModel):
...
+        "highlight": "highlight_field_1, highlight_field_2, ...",  
+        "highlightPreTag": "pre_tag",  
+        "highlightPostTag": "post_tag",  
+        "minimumCoverage": # (% of index that must be covered to declare query successful; default 100),  
+        "orderby": "orderby_expression",  
+        "queryType": "simple" (default) | "full",
+        "scoringParameters": [ "scoring_parameter_1", "scoring_parameter_2", ... ],  
+        "scoringProfile": "scoring_profile_name",  
+        "scoringStatistics" : "local" | "global",
+        "search": "simple_query_expression",
```
Comment (dev2049) [inline langchain/retrievers/azure_cognitive_search.py:49]: don't wan't search to be specified in here

**Review by dev2049 — COMMENTED** [review]
could we add a simple integration test? two small comments, otherwise looks good!

**Comment by leo-gan** [comment]
@matteo-campana Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**